src/components/settings/MicrosoftAdminOauthSettings.vue

```diff
@@ -0,0 +1,117 @@
+<!--
+  - @copyright 2022 Christoph Wurst <christoph@winzerhof-wurst.at>
+  -
+  - @author 2022 Christoph Wurst <christoph@winzerhof-wurst.at>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<template>
+	<form @submit.prevent="onSubmit">
+		<label for="mail-gmail-oauth-tenant-id"> {{ t('mail', 'Tenant ID') }} </label>
+		<input
+			id="mail-gmail-oauth-tenant-id"
+			v-model="tenantIdVal"
+			:disabled="loading"
+			type="text"
+			required>
+		<label for="mail-gmail-oauth-client-id"> {{ t('mail', 'Client ID') }} </label>
+		<input
+			id="mail-gmail-oauth-client-id"
+			v-model="clientIdVal"
+			:disabled="loading"
+			type="text"
+			required>
+		<label for="mail-gmail-oauth-client-secret"> {{ t('mail', 'Client secret') }} </label>
+		<input
+			id="mail-gmail-oauth-client-secret"
+			v-model="clientSecret"
+			:disabled="loading"
+			type="password"
+			required>
+		<button type="submit" :disabled="!tenantId || !clientIdVal || !clientSecret || clientSecret === PASSWORD_PLACEHOLDER || loading" class="primary">
```

Comment:
```suggestion
		<button type="submit" :disabled="!tenantIdVal || !clientIdVal || !clientSecret || clientSecret === PASSWORD_PLACEHOLDER || loading" class="primary">
```

Fixes save button never being enabled.

Reply:
Done